packages/front-end/components/Settings/EditDataSource/ExperimentAssignmentQueries/AddEditExperimentAssignmentQueryModal.tsx

```diff
@@ -97,6 +116,68 @@ export const AddEditExperimentAssignmentQueryModal: FC<EditExperimentAssignmentQ
     return null;
   }
 
+  const handleTestQuery = async () => {
+    setQueryError(null);
+    setQuerySuccessMessage(null);
+    setQueryWarnings([]);
+    setNoRowsReturned(false);
+
+    const value: ExposureQuery = {
+      name: exposureQuery.name,
+      query: userEnteredQuery,
+      id: dataSource.id,
+      userIdType: userEnteredUserIdType,
+      dimensions: userEnteredDimensions,
+      hasNameCol: userEnteredHasNameCol ? userEnteredHasNameCol : false,
+    };
+
+    try {
+      const requiredColumns = getRequiredColumns(value);
+      validateSQL(value.query, requiredColumns);
+
+      const res: TestQueryResults = await apiCall("/query/exposure/validity", {
+        method: "POST",
+        body: JSON.stringify({
+          query: value.query,
+          id: value.id,
+          requiredColumns,
+        }),
+      });
+
+      if (res.errorMessage) {
```

Comment:
Instead of setting all of these manually, can I just create 1 state for queryTestResult and make it the shape of null or QueryTestResult and then do conditional rendering? This would make it to where I didn't have to individually reset all of the states on each additional test